### PR TITLE
debug: generate coredump on netbsd.

### PR DESCRIPTION
**Pull request by devnexen**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)

### CODE REVIEW

**Review by ret2libc — CHANGES_REQUESTED** [review]
I think it's better to place a `#if` in rz_debug_native plugin, in the gcore method.

**Comment by devnexen** [comment]
would be ideal but it is to avoid changing the plugin api calls (ie here not using the buffer file type as we do not use the file descriptor).

File: librz/core/cmd_debug.c [hunk 1]
```diff
@@ -4830,6 +4831,11 @@ RZ_IPI int rz_cmd_debug(void *data, const char *input) {
 			} else {
 				perror("rz_buf_new_file");
 			}
+#else
+			if (ptrace(PT_DUMPCORE, core->dbg->pid, corefile, strlen(corefile)) == -1) {
+				eprintf("dg: coredump failed\n");
```
Comment (wargio) [inline librz/core/cmd_debug.c]: RZ_LOG_ERROR

**Review by wargio — CHANGES_REQUESTED** [review]

**Comment by ret2libc** [comment]
> would be ideal but it is to avoid changing the plugin api calls (ie here not using the buffer file type as we do not use the file descriptor).

That's a good point indeed. I'd still prefer having all the logic wrapped around in the right place though.
I see 2 options:
1) add a new `rz_buf_get_name` API to rz_buf that gets the name of a RzBuffer (for file-type RzBuf it would be the filename, for other things it can be less meaningful things....). Then use rz_buf_get_name inside gcore debug method to retrieve the filename and use it on NetBSD
2) change the debug plugin API to accept filename + RzBuffer and use the filename for netbsd. 

Probably the 2) is just 2 lines of change considering that only two plugins implement gcore. 1) would definitely be more involved. What do you think?

**Review by XVilka — APPROVED** [review]

**Review by ret2libc — CHANGES_REQUESTED** [review]
I think there's also a win dbg plugin, can you double check?

**Review by yossizap — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Pull request merged**